modules/indexer/code/indexer.go

```diff
@@ -286,22 +286,24 @@ func UpdateRepoIndexer(repo *models.Repository) {
 func populateRepoIndexer(ctx context.Context) {
 	log.Info("Populating the repo indexer with existing repositories")
 
-	exist, err := db.IsTableNotEmpty("repository")
+	engine := db.GetEngine(db.DefaultContext)
+	tableRepository := new(models.Repository)
+
+	exist, err := engine.Table(tableRepository).Exist()
```

Comment:
Please don't bring DB specific code in to modules.

Reply:
The code was there. The code was accessing database directly via `db.Xxx`.

The new code is just more clear than before.